### PR TITLE
Scripts/Events: Selina Dourman

### PR DESCRIPTION
**Pull request by meji46**
**Changes proposed:**

-  Added missing AI, POIs and conditions for Selina Dourman (thanks to @mdX7 for pointing me to use CAST_AI)

**Issues addressed:**
None

**Tests performed:**
Builds, tested in-game.

**Known issues and TODO list:** (add/remove lines as needed)
None

### CODE REVIEW

File: src/server/scripts/Events/darkmoon_faire.cpp [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+                CloseGossipMenuFor(player);
+                break;
+            default:
+                break;
+        }
+
+        return false;
+    }
+
+    void SetData(uint32 id, uint32 value) override
```
Comment (mdX7) [inline src/server/scripts/Events/darkmoon_faire.cpp:126]: you can move the entire code of theis function into the method which is being called by the areatrigger (in this case `SetTalkTarget`. You might want to rename it to something like `DoWelcomeTalk` or something when you've moved it there.

File: src/server/scripts/Events/darkmoon_faire.cpp [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+            default:
+                break;
+        }
+
+        return false;
+    }
+
+    void DoWelcomeTalk(ObjectGuid talkTargetGuid)
+    {
+        if (!talkTargetGuid.IsEmpty())
```
Comment (mdX7) [inline src/server/scripts/Events/darkmoon_faire.cpp:122]: No need to check that. ObjectAccessor takes care of this

File: src/server/scripts/Events/darkmoon_faire.cpp [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+        }
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        switch (_events.ExecuteEvent())
+        {
+            case EVENT_SELINA_TALK_COOLDOWN:
```
Comment (mdX7) [inline src/server/scripts/Events/darkmoon_faire.cpp:141]: This actually looks kinda weird. I'd set a variable _talkCooldown to false when timer runs out and set it to true when doing the talk

File: src/server/scripts/Events/darkmoon_faire.cpp (hunk 2 above)
Comment (meji46) [reply src/server/scripts/Events/darkmoon_faire.cpp]: Done!

**Pull request merged**